autonomic-administration-plugin/src/test/java/br/com/autonomiccs/autonomic/administration/plugin/AdministrationAgentTest.java

```diff
@@ -0,0 +1,665 @@
+
+import com.cloud.dc.ClusterVO;
+import com.cloud.dc.DataCenter.NetworkType;
+import com.cloud.dc.DataCenterVO;
+import com.cloud.host.HostVO;
+import com.cloud.utils.exception.CloudRuntimeException;
+import com.cloud.vm.UserVmService;
+import com.cloud.vm.VMInstanceVO;
+
+import br.com.autonomiccs.autonomic.administration.algorithms.ClusterAdministrationHeuristicAlgorithm;
+import br.com.autonomiccs.autonomic.administration.plugin.services.AutonomicClusterManagementService;
+import br.com.autonomiccs.autonomic.algorithms.commons.resources.CloudResources;
+import br.com.autonomiccs.autonomic.algorithms.commons.resources.ClusterResourcesUp;
+import br.com.autonomiccs.autonomic.algorithms.commons.resources.HostResources;
+import br.com.autonomiccs.autonomic.algorithms.commons.resources.VmResources;
+import br.com.autonomiccs.autonomic.algorithms.commons.services.CloudResourcesService;
+import br.com.autonomiccs.autonomic.algorithms.commons.services.ClusterResourcesService;
+import br.com.autonomiccs.autonomic.algorithms.commons.services.HostResourcesService;
+import br.com.autonomiccs.autonomic.plugin.common.services.AutonomicClusterManagementHeuristicService;
+import br.com.autonomiccs.autonomic.plugin.common.services.ClusterService;
+import br.com.autonomiccs.autonomic.plugin.common.services.HostService;
+import br.com.autonomiccs.autonomic.plugin.common.services.VirtualMachineService;
+import br.com.autonomiccs.autonomic.plugin.common.services.ZoneService;
+import br.com.autonomiccs.autonomic.plugin.common.utils.ThreadUtils;
+
+@RunWith(MockitoJUnitRunner.class)
+public class AdministrationAgentTest {
+
+    @Spy
+    @InjectMocks
+    private AdministrationAgent administrationAgent;
+    @Mock
+    private ClusterService clusterService;
+    @Mock
+    private AutonomicClusterManagementHeuristicService autonomicManagementHeuristicService;
+    @Mock
+    private ClusterAdministrationHeuristicAlgorithm clusterAdministrationHeuristicAlgorithm;
+    @Mock
+    private ZoneService zoneService;
+    @Mock
+    private AutonomicClusterManagementService autonomicClusterManagementService;
+    @Mock
+    private CloudResourcesService cloudResourcesService;
+    @Mock
+    private ClusterResourcesService clusterResourcesService;
+    @Mock
+    private HostService hostService;
+    @Mock
+    private VirtualMachineService virtualMachineService;
+    @Mock
+    private ThreadUtils threadUtils;
+    @Mock
+    private UserVmService userVmService;
+    @Mock
+    private HypervisorManager hypervisorManager;
+    @Mock
+    private HostResourcesService hostResourcesService;
+
+    @Before
+    public void setup() {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test
+    public void receiveClusterToBeManagedTestClusterIdNull() {
+        administrationAgent.receiveClusterToBeManaged(null);
+        verifyReceiveClusterToBeManaged(0, 0);
+    }
+
+    @Test
+    public void receiveClusterToBeManagedTestClusterIdNotNullButClusterIsNull() {
```

Comment:
I think the name of this method would be better:
receiveClusterToBeManagedTestClusterIdNotNullButClusterWasNotFound